### PR TITLE
Refactor groups

### PR DESCRIPTION
**Pull request by tillmonjesse**

### CODE REVIEW

File: src/js/factory/characterFactory.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import {WIDTH, HEIGHT} from '../constant.js';
-export const characterCreate = (scene, group, config, enemies, punchDirection, cursors) => {
+import {getAllies, getEnemies, groupIdentifier, enemyIdentifier} from '../service/groups.js';
+export const characterCreate = (scene, config, punchDirection, cursors) => {
+	let group = groupIdentifier(config) 
+	let enemies = enemyIdentifier(config)
 	let character = group.create(
```
Comment (Adam-Kaplan) [inline src/js/factory/characterFactory.js:6]: **[NBC]** could be written as 

```js
let character = groupIdentifier(config).create( /* */ )
```

File: src/js/scene/initial.js [hunk 2]
```diff
@@ -99,3 +89,7 @@ function update ()
     }
     
 }
+// make service for groups
```
Comment (Adam-Kaplan) [inline src/js/scene/initial.js:92]: Are these still needed?

File: src/js/service/groups.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export const getEnemies = () => {
+	return enemies;
+};
+export const getPlatform = () => {
+	return platform;
+};
+export const groupIdentifier = (config) => {
+	if (config.team === 'ally')
+	{
+		return getAllies()
```
Comment (Adam-Kaplan) [inline src/js/service/groups.js:27]: Please add `;`

File: src/js/service/groups.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	return allies;
+};
+export const getEnemies = () => {
+	return enemies;
+};
+export const getPlatform = () => {
+	return platform;
+};
+export const groupIdentifier = (config) => {
+	if (config.team === 'ally')
```
Comment (Adam-Kaplan) [inline src/js/service/groups.js:25]: Please export the strings `'ally'` and `'enemy'` as a constant from this file.

```js
export const ALLY_TEAM_NAME = 'ally';
```

Comment (tillmonjesse) [reply src/js/service/groups.js:25]: should these be imported into charaterfactory.js?

Comment (Adam-Kaplan) [reply src/js/service/groups.js:27]: To anywhere the constants are needed.

**Pull request merged**